firebase-authexchange/src/main/kotlin/com/google/firebase/authexchange/network/Routes.kt

```diff
@@ -0,0 +1,105 @@
+package com.google.firebase.authexchange.network
+
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+
+/**
+ * Defines what an HTTP request may look like when interacting with the backend API for
+ * [FirebaseAuthExchange][com.google.firebase.authexchange.FirebaseAuthExchange].
+ */
+@Serializable internal sealed interface Request
+
+/** Defines what an HTTP response may look like when submitting a [Request]. */
+@Serializable internal sealed interface Response
+
+/**
+ * OIDC Credential.
+ *
+ * Useful for apps with external login already set-up, otherwise known as a headless OIDC flow.
+ *
+ * @property idToken JWT encoded OIDC token returned from a third party provider
+ */
+@Serializable internal data class ImplicitCredentialsP(val idToken: String)
+
+/**
+ * Auth Token that can be used to access certain Firebase Services.
+ *
+ * This is merely the proto type for the network layer, and is completely separate from the user
+ * facing [AuthExchangeToken][com.google.firebase.authexchange.AuthExchangeToken].
+ *
+ * @property accessToken signed JWT containing claims that identify a user
+ * @property timeToLive the duration from the time this token is minted until its expiration
+ */
+@Serializable
+internal data class AuthExchangeTokenP(
+  val accessToken: String,
+  @SerialName("ttl") val timeToLive: String
+)
+
+/**
+ * Request header for the `/ExchangeInstallationAuthToken` endpoint.
+ *
+ * @see ExchangeTokenResponseP
+ *
+ * @property token relative resource name of the audience project and location
+ * @property installationAuthToken the installation token issued to the app
+ */
+@Serializable
+internal data class ExchangeInstallationAuthTokenRequestP(
+  val token: String,
```

Comment:
The `token` field is unnecessary as part of the request object (here and in all the requests). In the backend, the `token` field is extracted from the request URL.

Reply:
Ahh I see. Is it the `key` param, or is it actually called `token`?

Reply:
What do you mean by `key` param? For additional context -- here is the [request path](http://google3/google/firebase/authexchange/v1alpha/token_exchange_service.proto;l=35;rcl=492327771). Note the `token=` portion of the request path -- that is where the `token` field is populated from.